contracts/FractalUsul.sol

```diff
@@ -1,86 +1,469 @@
         bytes memory data,
         Transaction[] calldata transactions,
         string calldata title,
         string calldata description,
         string calldata documentationUrl
-  ) external {
-      require(
-          isStrategyEnabled(strategy),
-          "voting strategy is not enabled for proposal"
-      );
-      require(transactions.length > 0, "proposal must contain transactions");
-
-      bytes32[] memory txHashes = new bytes32[](transactions.length);
-      for (uint256 i = 0; i < transactions.length; i++) {
-        txHashes[i] = getTransactionHash(
-          transactions[i].to, 
-          transactions[i].value,
-          transactions[i].data, 
-          transactions[i].operation
-        );
-      }
-
-      proposals[totalProposalCount].txHashes = txHashes;
-      proposals[totalProposalCount].strategy = strategy;
-      IStrategy(strategy).receiveProposal(
-          abi.encode(totalProposalCount, txHashes, data)
-      );
-      emit ProposalCreated(strategy, totalProposalCount, msg.sender);
-      emit ProposalMetadataCreated(
-        totalProposalCount, 
-        transactions, 
-        title, 
-        description, 
-        documentationUrl
-      );
-      totalProposalCount++;
-  }
-
-  /// @notice Gets the transaction hashes associated with a given proposald
-  /// @param proposalId The ID of the proposal to get the tx hashes for
-  /// @return bytes32[] The array of tx hashes
-  function getProposalTxHashes(uint256 proposalId) external view returns (bytes32[] memory) {
-    return proposals[proposalId].txHashes;
-  }
-}
\ No newline at end of file
+    ) external {
+        require(
+            isStrategyEnabled(strategy),
+            "Voting strategy is not enabled"
+        );
+        require(transactions.length > 0, "Proposal must contain at least one transaction");
+        require(IBaseStrategy(strategy).isProposer(msg.sender), "Caller cannot submit proposals");
+
+        bytes32[] memory txHashes = new bytes32[](transactions.length);
+        for (uint256 i = 0; i < transactions.length; i++) {
+            txHashes[i] = getTxHash(
+                transactions[i].to,
+                transactions[i].value,
+                transactions[i].data,
+                transactions[i].operation
+            );
+        }
+
+        proposals[totalProposalCount].txHashes = txHashes;
+        proposals[totalProposalCount].strategy = strategy;
+        IBaseStrategy(strategy).receiveProposal(
+            abi.encode(totalProposalCount, txHashes, data)
+        );
+        emit ProposalCreated(
+            strategy,
+            totalProposalCount,
+            msg.sender,
+            transactions,
+            title,
+            description,
+            documentationUrl
+        );
+
+        totalProposalCount++;
+    }
+
+    /// @notice Cancels an array of proposals
+    /// @param proposalIds Array of proposals to cancel
+    function cancelProposals(uint256[] memory proposalIds) external onlyOwner {
```

Comment:
is this function even necessary?

I'm trying to think of a situation where a DAO would create a proposal to cancel another proposal, would that even work?

Reply:
it seems like you can just vote it down instead

Reply:
how does it work if there is more than one strategy enabled? You can submit a proposal to either strategy?

Reply:
Ok cool, will remove

Reply:
Removed this function